### PR TITLE
Adds maintenance mode command.

### PR DESCRIPTION
**Pull request by Nonsense331**

### CODE REVIEW

File: lib/subspace/commands/maintenance_mode.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    run
+  end
+
+  def run
+    on_off = @options.__hash__[:on] ? "on" : "off"
+    ansible_options = ["--diff", "-e maintenance_hosts=#{@hosts}", "--tags=maintenance_#{on_off}"]
+    PASS_THROUGH_PARAMS.each do |param_name|
+      x = param_name.split('-')[1..-1].map(&:upcase).join('_')
+      hash_key = (param_name.gsub('-', '_') + (x == '' ? '' : "_#{x}")).to_sym
+      value = @options.__hash__[hash_key]
```
Comment (kyla-harper) [inline lib/subspace/commands/maintenance_mode.rb]: That's an interesting way to be able to access data.

Comment (boatrite) [reply lib/subspace/commands/maintenance_mode.rb]: P sure this same code is used in a few of these classes in the same way. Time to refactor?

Comment (Nonsense331) [reply lib/subspace/commands/maintenance_mode.rb]: @boatrite :didit:

Comment (boatrite) [reply lib/subspace/commands/maintenance_mode.rb]: haha noice.

File: lib/subspace/commands/base.rb [hunk 2]
```diff
@@ -53,6 +53,23 @@ def confirm_overwrite(file_path)
         return answer.downcase.start_with? "y"
       end
 
+      def pass_through_params
+        ansible_options = []
+        self.class::PASS_THROUGH_PARAMS.each do |param_name|
```
Comment (boatrite) [inline lib/subspace/commands/base.rb]: fancy, i wasn't sure how this could be done :+1: i didn't realize this was valid syntax lol

**Pull request merged**